### DISCUSSION TITLE
Create stacked AND grouped columns

### DISCUSSION THREAD

**Discussion by nicolasthemmer-aaa** [patrykandpatrick/vico#660] [Ideas]
Hey everyone! 

I have two data sets that need to be displayed in the same chart. That wouldnt be an issue with the merge mode GROUPED. But both columns need to be stacked as well.. I can only see the option the choose either or (stacked or grouped). Is there any way to achieve that? 

Maybe I will have to use GROUPED as merge mode. But than I would need to color each segment of the column based on the y-value. Any ideas?

![image](https://github.com/patrykandpatrick/vico/assets/13311585/8e7fb024-6e18-48b3-bf29-464a4357b970)

**Comment by nicolasthemmer-aaa** [comment]
For anyone else having the same issue:

I was able to achieve my result by overriding the `drawInternal` in the `ColumnCartesianLayer`. By default I was using `MergeMode.Stacked` (all columns got stacked) and then I created my own `getDrawingStart `function to shift the first part of the stacked series to the left and the second part to the right (inside of the `drawChartInternal ` `model.series.foreachIndexed`)

Also don't forget to clear the `stackInfo` after shifting the first part of the series. That will reset the `bottomY` for the second part of the series.

**Reply by rickardzettervall** [reply]
Thank you, this works for now.

@patrickmichalik, is support of combined stack and group columns planned for 2.0?

**Reply by Gowsky** [reply]
Hello, @rickardzettervall. This isn't planned for Vico 2 at the moment. We'll look into adding such functionality if there's more interest in it. I'm reopening this discussion and moving it to "Ideas."
